extensions/amp-ad-exit/0.1/filters/click-location.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FilterType, Filter} from './filter';
+import {user} from '../../../../src/log';
+
+export class ClickLocationFilter extends Filter {
+  /**
+   * @param {string} name The user-defined name of the filter.
+   * @param {!../config.ClickLocationConfig} spec
+   */
+  constructor(name, spec, win) {
+    super(name);
+    user().assert(isValidClickLocationSpec(spec), 'Invaid ClickLocation spec');
+
+    /** @private {number} */
+    this.leftBorder_ = spec.left || 0;
+
+    /** @private {number} */
+    this.rightBorder_ = spec.right || 0;
+
+    /** @private {number} */
+    this.topBorder_ = spec.top || 0;
+
+    /** @private {number} */
+    this.bottomBorder_ = spec.bottom || 0;
+
+    /** @private {string|undefined} */
+    this.relativeTo_ = spec.relativeTo;
+
+    /** @private {!Window} */
+    this.win_ = win;
+  }
+
+  /** @override */
+  filter(event) {
+    let left = 0;
+    let top = 0;
+    let right = this.win_./*REVIEW*/innerWidth;
+    let bottom = this.win_./*REVIEW*/innerHeight;
+    if (this.relativeTo_) {
+      const relativeElement = this.win_.document.querySelector(
+          this.relativeTo_);
+      user().assert(relativeElement,
+          `relativeTo element ${this.relativeTo_} not found.`);
+      const elementRect = relativeElement./*REVIEW*/getBoundingClientRect();
```

Comment:
Sure, I can add some doc about it.
The border protection filter is to filter the clicks which are on the border of the ads. The border area can be relative the current iframe view port or relative to an element which is specified by relativeTo in the config.
Can we make an agreement so that I can change the implementation accordingly?